tests/e2e/mapper/bluetooth/bluetooth_test.go

```diff
@@ -0,0 +1,297 @@
+			expectedSchedule = []scheduler.Schedule{schedule}
+			scheduleCreateTopic1 := "$ke/device/bluetooth-mapper/mock-temp-sensor-instance/scheduler/create"
+			scheduleCreate, err := json.Marshal(expectedSchedule)
+			if err != nil {
+				common.Failf("Error in marshalling: %s", err)
+			}
+			TokenClient = Client.Publish(scheduleCreateTopic1, 0, false, scheduleCreate)
+			if TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("Publish to Topic  Failed  %s, %s", TokenClient.Error(), scheduleCreateTopic1)
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+		})
+		It("E2E_DATACONVERTER PROPERLY EXECUTED: Test whether dataconverter has performed conversion correctly", func() {
+			Eventually(func() bool {
+				return dataConverted
+			}, "30s", "0.5s").Should(Equal(true))
+		})
+		AfterEach(func() {
+			Client.Disconnect(1)
+			common.PrintTestcaseNameandStatus()
+		})
+	})
+
+	// Checking whether schedule executed the expected number of times using runtime update
+	Context("Test whether schedule is executed properly", func() {
+		BeforeEach(func() {
+			// Subscribing to topic where scheduler publishes the data
+			ClientOpts = helpers.HubClientInit(ctx.Cfg.MqttEndpoint, "bluetoothmapper", "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if TokenClient = Client.Connect(); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Connect() Error is %s", TokenClient.Error())
+			} else {
+				common.Info("Subscribe Connection successful")
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+			scheduletopic := "$ke/device/bluetooth-mapper/mock-temp-sensor-instance/scheduler/result"
+			Token := Client.Subscribe(scheduletopic, 0, ScheduleExecute)
+			if Token.Wait() && TokenClient.Error() != nil {
+				common.Failf("Subscribe to Topic  Failed  %s, %s", TokenClient.Error(), scheduletopic)
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+			var expectedSchedule []scheduler.Schedule
+			// Create and publish run time data for scheduler
+			var schedule scheduler.Schedule
+			schedule.Name = "temperature"
+			schedule.Interval = 3000
+			schedule.OccurrenceLimit = 10
+			schedule.Actions = []string{"IRTemperatureData"}
+			expectedSchedule = []scheduler.Schedule{schedule}
+			scheduleCreateTopic := "$ke/device/bluetooth-mapper/mock-temp-sensor-instance/scheduler/create"
+			scheduleCreate, err := json.Marshal(expectedSchedule)
+			if err != nil {
+				common.Failf("Error in marshalling: %s", err)
+			}
+			TokenClient = Client.Publish(scheduleCreateTopic, 0, false, scheduleCreate)
+			if TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("Publish to Topic  Failed  %s, %s", TokenClient.Error(), scheduleCreateTopic)
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+		})
+		It("E2E_SCHEDULE PROPERLY EXECUTED: Test whether the schedule is executed the specified number of times", func() {
+			timesSpecified = 10
+			time.Sleep(40 * time.Second)
```

Comment:
can we reduce the schedule intevel to maybe once in every 1 second and schedule times to 5 ? So that the test can wait for 8 or 10 seconds..... 40 seconds is a big delay

Reply:
Updated